### PR TITLE
Feature/redesign faq

### PR DESCRIPTION
**Pull request by ChristianBay**
Open point is the location for the JSON files of the FAQ block (e.g. faq_allgemein.json). 
In my opinion this content should be loaded from a static file (and not from a file inside the git repo)

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -37,14 +38,16 @@ import { AuthenticationPageComponent } from './authentication-pages/components/a
...
+import { FaqBlockComponent } from './faq-block/faq-block.component';
 
 // eslint-disable-next-line prefer-arrow/prefer-arrow-functions
 export function createTranslateLoader(http: HttpClient) {
   return new TranslateHttpLoader(http, './assets/i18n/', '.json');
 }
 
 @NgModule({
-  declarations: [AppComponent, MainLayoutComponent, FooterOnlyLayoutComponent, AboutComponent],
+  declarations: [AppComponent, MainLayoutComponent, FooterOnlyLayoutComponent, AboutComponent, FaqPageComponent, FaqBlockComponent],
```
Comment (SandroGlueck) [inline src/app/app.module.ts:50]: To stick with the rest of the code you could put these two components into a separate `FaqModule`.
That way everything regarding FAQ is encapsulated into it's own module and the main `AppModule` only imports all the "feature modules"

File: src/app/faq-block/faq-block.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  <h2>{{headline}}</h2>
+  <div class="questionsAnswers">
+    <div *ngFor="let item of faq;let index=index;">
+      <div class="collapsible-toolbar">
+        <span class="collapsible-toolbar__title question">{{item.question}}</span>
+        <span class="collapsible-toolbar__toggle" (click)="toggle[index] = !toggle[index]">{{toggle[index] ? 'ausblenden' : 'einblenden'}}</span>
+        <i class="far" [ngClass]="toggle[index] ? 'fa-chevron-down' : 'fa-chevron-right'"></i>
+      </div>
+      <div *ngIf="toggle[index]" id="faqAnswer{{index+1}}" class="answer" [innerHTML]="item.answer"></div>
+    </div>
```
Comment (SandroGlueck) [inline src/app/faq-block/faq-block.component.html:11]: I would separate this into an own `FaqItemComponent`.
That way you could have a `isToggled` property for each individual item, instead of using this (for me) confusing array of booleans.

File: src/app/faq-block/faq-block.component.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  selector: 'app-faq-block',
+  templateUrl: './faq-block.component.html',
+  styleUrls: ['./faq-block.component.scss']
+})
+export class FaqBlockComponent implements OnInit {
+
+  @Input() headline: string;
+  @Input() json: string;
+
+  toggle = [];
```
Comment (SandroGlueck) [inline src/app/faq-block/faq-block.component.ts:18]: If you stick with the current implementation and keep this array, please specify the type of values that are supposed to be stored in the array -> `toggle: boolean[] = []`.

File: src/app/faq-block/faq-block.component.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<div class="faqblock">
+  <h2>{{headline}}</h2>
+  <div class="questionsAnswers">
```
Comment (SandroGlueck) [inline src/app/faq-block/faq-block.component.html:3]: Couldn't find this class anywhere - please remove it if it doesn't exist.

File: src/app/app-routing.module.ts [hunk 5]
```diff
@@ -114,6 +115,7 @@ const routes: Routes = [
   { path: 'my-file', component: MyFileComponent, canActivate: [AuthGuard] },
   { path: 'issues', component: IssuesComponent },
   { path: 'about', component: AboutComponent },
+  { path: 'faq', component: FaqPageComponent },
```
Comment (SandroGlueck) [inline src/app/app-routing.module.ts:118]: If you move the route into the child section of the `MainLayoutComponent`, you can omit stuff like the <app-header> in your `FaqPageComponent`

File: src/app/faq-page/faq-page.component.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<div class="faq-page standard-content-container-width">
+  <app-header></app-header>
```
Comment (SandroGlueck) [inline src/app/faq-page/faq-page.component.html:2]: See comment on app.routing.module.ts on how to omit <app-header>, <app-footer> and such.

File: src/app/faq-page/faq-page.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+import { FaqPageComponent } from './faq-page.component';
+
+describe('FaqPageComponent', () => {
+  let component: FaqPageComponent;
+  let fixture: ComponentFixture<FaqPageComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ FaqPageComponent ]
```
Comment (SandroGlueck) [inline src/app/faq-page/faq-page.component.spec.ts:11]: Tests currently fail due to missing imports and declarations.

File: src/app/faq-block/faq-block.component.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+  faq: Faq[];
+
+  constructor() {
+  }
+
+  ngOnInit(): void {
+    import('src/assets/data/' + this.json).then(data => {
```
Comment (SandroGlueck) [inline src/app/faq-block/faq-block.component.ts:26]: Maybe you can once more check if you can use the HttpClient approach instead of `import` to load the json. This would make it easier to switch to statically served files (e.g. through S3) later down the road.

File: src/app/faq-page/faq-page.component.html (hunk 6 above)
Comment (SandroGlueck) [reply src/app/faq-page/faq-page.component.html:2]: This also leads to the header and footer not taking up the entire width

image

File: src/app/faq-block/faq-block.component.ts (hunk 8 above)
Comment (SandroGlueck) [reply src/app/faq-block/faq-block.component.ts:26]: In order to load the json correctly from the asset folder, you will most likely only have to remove the `src` from the path.

File: tsconfig.json [hunk 9]
```diff
@@ -3,6 +3,8 @@
   "compilerOptions": {
     "baseUrl": "./",
     "outDir": "./dist/out-tsc",
+    "resolveJsonModule": true,
+    "allowSyntheticDefaultImports": true,
```
Comment (SandroGlueck) [inline tsconfig.json:7]: If you change to the `HttpClient` approach, please undo these changes.

**Review by SandroGlueck — CHANGES_REQUESTED** [review]
Great work - I only have a few general points that would improve the structure of components/modules.

Biggest "problem" at the moment is, that the tests automatically generated by Angular fail, due to missing imports in the TestBed in the `*.spec.ts` files.

**Comment by ChristianBay** [comment]
> Great work - I only have a few general points that would improve the structure of components/modules.
> 
> Biggest "problem" at the moment is, that the tests automatically generated by Angular fail, due to missing imports in the TestBed in the `*.spec.ts` files.

Fixed

File: tsconfig.json (hunk 9 above)
Comment (ChristianBay) [reply tsconfig.json]: Done

File: src/app/faq-block/faq-block.component.ts (hunk 8 above)
Comment (ChristianBay) [reply src/app/faq-block/faq-block.component.ts]: Done

File: src/app/faq-page/faq-page.component.spec.ts (hunk 7 above)
Comment (ChristianBay) [reply src/app/faq-page/faq-page.component.spec.ts]: Done

File: src/app/faq-page/faq-page.component.html (hunk 6 above)
Comment (ChristianBay) [reply src/app/faq-page/faq-page.component.html]: Done

File: src/app/faq-block/faq-block.component.html (hunk 4 above)
Comment (ChristianBay) [reply src/app/faq-block/faq-block.component.html]: Done

File: src/app/faq-block/faq-block.component.ts (hunk 3 above)
Comment (ChristianBay) [reply src/app/faq-block/faq-block.component.ts]: It was black magic ... 
I have now created an additional FAQ entry with the "show" attribute

File: src/app/faq-block/faq-block.component.html (hunk 2 above)
Comment (ChristianBay) [reply src/app/faq-block/faq-block.component.html]: Done

File: src/app/app.module.ts (hunk 1 above)
Comment (ChristianBay) [reply src/app/app.module.ts]: Done
